lib/web/fetch/index.js

```diff
@@ -120,12 +122,66 @@ class Fetch extends EE {
   }
 }
 
-// https://fetch.spec.whatwg.org/#fetch-method
-function fetch (input, init = undefined) {
-  webidl.argumentLengthCheck(arguments, 1, 'globalThis.fetch')
+// This will publish all diagnostic events only when we have subscribers.
+function ifSubscribersRunStores (request, input, init, callback) {
+  const hasSubscribers = subscribersCheck()
 
-  // 1. Let p be a new promise.
-  const p = createDeferredPromise()
+  if (hasSubscribers) {
+    const context = { request, input, init, result: null, error: null }
```

Comment:
```suggestion
    const context = { request, input, init: init ?? {}, result: null, error: null }
```

Reply:
So if init is null it sets to {}?

Reply:
undefined and null are converted to {} by the webidl spec